app/javascript/components/search/StudySearchProvider.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import { fetchSearch } from 'lib/scp-api'
+import _cloneDeep from 'lodash/cloneDeep'
+import _assign from 'lodash/assign'
+
+const emptySearch = {
+  params: {
+    terms: '',
+    facets: {}
+  },
+  results: [],
+  isLoaded: false,
+  isError: false,
+  updateSearch: () => { throw new Error('You are trying to use this context outside of a Provider container') }
+}
+
+export const StudySearchContext = React.createContext(emptySearch)
+
+export default class StudySearchProvider extends React.Component {
+  constructor(props) {
+    super(props)
+    let initialState = _cloneDeep(emptySearch)
+    initialState.updateSearch = this.updateSearch
+    this.state = initialState
+  }
+
+  componentDidMount() {
+    this.updateSearch();
+  }
+
+  updateSearch = async (searchParams) => {
+    const effectiveParams = _assign(this.state.params, searchParams)
```

Comment:
I prefer standard JavaScript when an equivalent exists, e.g. `Object.assign` here.  But I consider this a matter of taste.

Reply:
concur